### DISCUSSION TITLE
Some basic protect listener like the EssentialsProtect plugin.

### DISCUSSION THREAD

**Discussion by sakurawald** [Zrips/CMI#7242] [Feature Requests]
Just like the EssentialsProtect plugin, there are some functions that CMI still don't have.
But this function is very basic and useful, more detailed protect than miniecraft:gamerule command.
like: preventCreeperExplosion, preventWitherExplosion, preventWitherBlockDamage, preventTNTExplosion, preventFireSpread etc.
There are many protections optionally in EssentialsProtect.

**Comment by mrfloris** [comment]
I use Worldguard personally to deal with world protections, they have global and per region flags, spawn control, and blacklisted items, and member and non-member region control - and all on a per world basis. It is much more detailed than CMI would be able to do out of the box. 

I feel that it would be awesome if I didnt' have to use worldguard, haha. But I feel full on world protection like that is kinda out of scope for CMI. 

Just my personal opinion.

**Reply by sakurawald** [reply]
I use WorldGuard in the past, but the most big problem is PERFORMANCE.
since i have use the Residence created by Zrips.
The 2 plugin are very similar. And i don't want to use some other plugins to solve this problem.
So i just beleive Zrips can add the BASIC PROTECTION FUNCTION inside CMI, at least EssentialsProtect will have these functions.

**Comment by mrfloris** [comment]
technically essentialsx doesn't have it, EssentialsProtect is an addon for essentials, there's no addon for cmi called cmiProtection. hence why I replied how I did. But it's a fair request to make, no worries. I don't make the decisions here ;) don't worry - just giving my own input on this. 

Doesn't residence have those anti grief and flags as protection build in btw? (I don't know, I use wg on the servers anyway for theother worlds that don't use residence) <https://www.zrips.net/residence/antigrief/> / <https://www.zrips.net/residence/flags/>

**Reply by sakurawald** [reply]
Yeah, Residence did have some flags to set every protection, but you can't create a world-wide residence.
and its config did have some anti-grief module, but not very detailed like EssentialsProtect.
CMI has some more interesting and funny functions, but it seems lacks some basic functions like Essentials.

**Reply by mrfloris** [reply]
Ah, like a `_global_` with flags set to 'no break/place' - or how essentials does it.
ok, I see what you mean.
